src/modules/job-manager/annotate.c

```diff
@@ -145,19 +146,27 @@ void annotate_handle_request (flux_t *h,
     }
     if (annotations_update (ctx->h, job, annotations) < 0)
         goto error;
+    if (job->annotations) {
+        if (!(tmp = json_deep_copy (job->annotations)))
+            goto nomem;
+    }
     if (event_job_post_pack (ctx->event,
                              job,
                              "annotations",
                              EVENT_JOURNAL_ONLY,
                              "{s:O?}",
-                             "annotations", job->annotations) < 0)
+                             "annotations", tmp) < 0)
         goto error;
```

Comment:
This might not be a comment that should be addressed in this PR, but should the annotation event be a snapshot of all annotations?  It seems like it should be for a specific annotation.  The journal consumer would then apply each event and build the same snapshot.  That would seem more consistent with the information that other events carry.  For example, we don[t tell you the current job state in a state transition.  You have to apply each event and implement the state machine to recreate the state from the events.

Reply:
I assume you mean a snapshot of all annotations for the job?  Seems like a good idea. Perhaps should be rolled in for the discussion in #3081, as part of the general "persistence" of annotations.

Reply:
To be clear, my idea was that an annotation event would _not_ be a snapshot.  Instead it would be something along the lines of a single key-value pair.

Reply:
oh i misread your comment.  I see your point now, effectively each event in the journal should be the "delta" from the prior annotations.  That does make sense.  Perhaps a cleanup for a follow up PR.